### DISCUSSION TITLE
Get the parent field in a nested document field resolver

### DISCUSSION THREAD

**Discussion by nivo33** [MichalLytek/type-graphql#967] [Q&A] [answered]
Hi, I need to access the fields of the parent in a field resolver. @Root() used to contain a `$parent` field, but I just updated from mongoose ~5.10 to ~5.13 and its gone. Any ideas?

Here is my classes (simplified):

```
ObjectType()
class Entity { 
@Field()
name: string

@Field()
data: Data
}

ObjectType()
class Data{
@Field()
type: string
}

@Resolver(() => Data)
class BResolvers {
   @FieldResolver(() => string)
   metadata(@Root() root, @Ctx() ctx) {
    // I need to access the parents Entity.name field here, but I can't. `root` contains only the Data  fields.
   // used to be that root.$parent contained the fields of the parent Entity - its now gone, and root is just an instance of Data
   }
 
}
```

I have a simple resolver that returns an instance of Entity.

Thanks!

**Comment by oliversalzburg** [comment, marked as answer]
The root object is just what it is. If it did have a `$parent` property before you upgraded mongoose, then the previous version of mongoose added it. This doesn't sounds related to TypeGraphQL.

**Reply by nivo33** [reply]
Thanks. Solved it.
If anyone ever somehow encounters this same issue with this same setup...
The solution was accessing the field through `$parent()` and not `$parent`. This was changed somewhere in ~mongoose 5.11.x.
